@xen-orchestra/lite/src/composables/array-history.composable.md

```diff
@@ -0,0 +1,41 @@
+# useArrayHistory composable
+
+This composable allows you to keep a history of each removed item of an array.
+
+## Usage
+
+```typescript
+const myArray = ref([]);
+const history = useArrayHistory(myArray)
+
+myArray.push('A'); // myArray = ['A']; history = []
+myArray.push('B'); // myArray = ['A', 'B']; history = []
+myArray.shift(); // myArray = ['B']; history = ['A']
+```
+
+You can limit the number of items to keep in history:
+
+```typescript
+const myArray = ref([]);
+const history = useArrayHistory(myArray, 30);
+```
+
+Be careful when using an array of objects which is likely to be replaced (instead of being altered):
+
+```typescript
+const myArray = ref([]);
+const history = useArrayHistory(myArray);
+myArray.value = [{ id: 'foo' }, { id: 'bar' }];
+myArray.value = [{ id: 'bar' }, { id: 'baz' }]; // history = [{ id: 'foo' }, { id: 'bar' }]
+```
+
+In this case, `{ id: 'bar' }` is detected as removed since in JavaScript `{ id: 'bar' } !== { id: 'bar' }`.
+
+You must therefore use an identity function as third parameter to return the value to be used to detect deletion:
+
+```typescript
+const myArray = ref<{ id: string }[]>([]);
+const history = useArrayHistory(myArray, undefined, (item) => item.id);
+myArray.value = [{ id: 'foo' }, { id: 'bar' }];
+myArray.value = [{ id: 'bar' }, { id: 'baz' }]; // history = [{ id: 'foo' }]
```

Comment:
I find that weird that in this case, only `id: 'foo'` is in the history.
We are assigning a new array, that mean the history must be equal to the overwritten array, and not only to the difference between the old and new array.

Or maybe the composables need to be renamed to avoid confusion. `arrayDiffHistory` or something like that?

Reply:
Yes, maybe the name could be better.

It's indeed not a history of the array, but a **history of items being removed from the array** 🤔 

`arrayDiffHistory` would suggest it keeps an history of diffs (added/removed).

Maybe something more like `useArrayRemovalHistory`? Or `useArrayRemovedItemHistory`? (or `useHistoryOfItemsBeingRemovedFromTheArray` ? 🤯 😀  )

@pdonias @Rajaa-BARHTAOUI Do you have a better idea for the composable name?

Reply:
I'm not sure I understand @MathieuRA's point of view. It tracks the items that were removed from the array. `{ id: 'bar' }` hasn't been removed from the array so why should it be in the history?

I agree the name could be more explicit about what the feature does, though, and I think `useArrayRemovedItemsHistory` (plural) is explicit enough.